src/com/company/M9/MyHashMap.java

```diff
@@ -0,0 +1,105 @@
+package com.company.M9;
+
+public class MyHashMap<K, V> {
+
+    public MyHashMap.Node currentLastNode;
```

Comment:
I left several links for you to understand how hashMap works. But I don't see any significant changes.
HashMap should contain the `array` of buckets, every bucket is a linked list that consist of Node<K, V> elements.